typing/typecore.ml

```diff
@@ -1001,31 +1001,37 @@ let check_scope_escape loc env level ty =
 (* type_pat propagates the expected type as well as maps for
    constructors and labels.
    Unification may update the typing environment. *)
-(* constrs <> None => called from parmatch: backtrack on or-patterns
+(* map <> None => called from parmatch: backtrack on or-patterns
    explode > 0 => explode Ppat_any for gadts *)
-let rec type_pat ?(exception_allowed=false) ~constrs ~labels ~no_existentials
-          ~mode ~explode ~env sp expected_ty k =
+type map =
```

Comment:
No.

Reply:
Agreed, I propose `pattern_names_map` which I used in #8969, what do you think?